dask/dataframe/io/parquet/core.py

```diff
@@ -337,22 +359,38 @@ def read_parquet(
     return new_dd_object(graph, output_name, meta, divisions)
 
 
-def read_parquet_part(fs, func, meta, part, columns, index, kwargs):
+def check_multi_support(engine):
+    # Helper function to check that the engine
+    # supports a multi-partition read
+    return hasattr(engine, "multi_support") and engine.multi_support()
```

Comment:
Have you written this for fastparquet?

By the way, time tests on the original problem:

```
%time dd.read_parquet(path, engine="pyarrow").compute()
2min43
%time dd.read_parquet(path, engine="fastparquet").compute()
2min27
%time fastparquet.ParquetFile(path).to_pandas()
28s
```
Where the last is effectively skipping concat, as proposed here, but is (of course) single-threaded. I have considered whether fastparquet should use threaded dask internally for such situations (would require releasing the GIL!).

For comparison
```
%time pyarrow.read_pandas.read_pandas(path)  # output ordering is different to other 3
30s
```

Reply:
Note that main branch fastparquet contains syntactic nicety:
```
pf = ParquetFile(...)
pf2 = pf[1:4] # selects some row groups by index
```
and that row-filtering *is coming* sometime soonish.

Reply:
No - This PR doesn't enable the multi-file read for fastparquet.  This is mostly because my personal motivation is to expose the improved chunksize behavior in the arrow engine so it can be leveraged in dask_cudf.

I can also add it for fastparquet if ParquetFile can handle a file list (which I believe it does).

Reply:
It does. I think it should be added. I can release fastparquet if necessary.